examples/examples/file-writer/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use async_std::{fs::File, io::WriteExt, sync::Mutex};
+use prost::Message as pMessage;
+use zenoh_flow::{anyhow, prelude::*};
+
+#[export_sink]
+pub struct FileWriter {
+    input: Input<String>,
+    file: Mutex<File>,
+}
+
+#[async_trait::async_trait]
+impl Node for FileWriter {
+    async fn iteration(&self) -> Result<()> {
+        let (message, _) = self.input.recv().await?;
+
+        if let Message::Data(greeting) = message {
+            let mut file = self.file.lock().await;
+            file.write_all(greeting.as_bytes())
+                .await
+                .map_err(|e| zferror!(ErrorKind::IOError, "{:?}", e))?;
+            return file
+                .flush()
+                .await
+                .map_err(|e| zferror!(ErrorKind::IOError, "{:?}", e).into());
+        }
+
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl Sink for FileWriter {
+    async fn new(
+        _context: Context,
+        configuration: Option<Configuration>,
+        mut inputs: Inputs,
+    ) -> Result<Self> {
+        let file_path = match &configuration {
+            Some(conf) => conf["file"]
+                .as_str()
+                .ok_or_else(|| zferror!(ErrorKind::InvalidState))?,
+            None => "/tmp/greetings.txt",
+        };
+
+        Ok(FileWriter {
+            file: Mutex::new(
+                File::create(file_path) //File::create("/tmp/greetings.txt")
+                    .await
+                    .expect("Could not create '/tmp/greetings.txt'"),
```

Comment:
Use an `unwrap_or_else` + `panic!` to be able to give the name of the file that could not be created. It is not always "/tmp/greetings.txt" so always printing this file (especially when we don't want to it) is confusing.